### PR TITLE
#11867: Cleanup Unary composites

### PR DESCRIPTION
**Pull request by ayerofieiev-tt**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/11867

### Problem description
Amount of code written here is remarkably unnecessary

1. we declare structs to register as TT-NN ops (A)
2. we want to save some code and we can templates so we create structs templated with enum representing op type
3. we then create an actual function
4. we wrap each function into a structure to map function to enum (we create a structure which we wanted to avoid at step 1) (B)
5. we call B from A
6. we reregister A as ttnn-op

### What's changed
Clean things up

### Checklist
- [ ] Post commit CI [passes](https://github.com/tenstorrent/tt-metal/actions/runs/10565903813)

### CODE REVIEW

**Review by eyonland — APPROVED** [review]
So now we simply register the function directly with register_operation_from_function and stop using the template enum in the template args?

**Comment by ayerofieiev-tt** [comment]
> So now we simply register the function directly with register_operation_from_function and stop using the template enum in the template args?

I did consider that way, I see some artifact from that in the PR. 
But no, this is not whats going on in this change and I expect that the change is a bit controversial.

Here are my considerations:
First of all, I just wanted us to cleanup the situation around enum and handle struct.
I think it was supposed to reduce amount of code, but it increased it. And I think for no reason.
I described in the PR description why I think so.

Second, I saw three options. Each option is ok and just has some pros and cons, largely taste.

### 1 Register function directly
This approach is clean and I overall like it.
* Whatever happens to the registration - operation implementation does not have to be touched. Nice decoupling [great]
* There is no code duplication, e.g. I don't have to maintain interface of a function, of Operation::invoke, etc [great]
* Function is available to anyone and can be called directly, not going though the registered path [rather bad]
* Raw function can be called to compose whatever without paying any cost a decorator might add [rather good]
* Does not currently allow for overload, but there is no need for this today [neutral]
* Does not extend well if we need to associate more things with an operation, so might have to rewrite to a struct later [neutral]

### 2 Create struct, register struct, call to those functions from struct's invoke
Same as above, but because this is just a manual implementation - remove the point about code duplication.
Between 1 and 2 I definitely would prefer 1. 
Unless there is more to an operation than a function and something else has to be added. In the past it was a schema, it is not removed. Overloads, can probably be handled with more template code with option 1, though maybe not as conveniently. 

### 3 Create structs, register structs, move function impl inside operation::invoke impl
This removes the thing I like - the raw function. 
Means there is no nice separation. Registration infra (op struct) is tightly coupled with the math implementation itself.
Means when you compose, you might pay price of decorators. 
But it keeps things largely consistent. As far as I see there are no function registrations today. 


Because my main objective here was to minimize the maintenance surface, I went with option 3 as it removes much code and does not introduce a new way of doing things.

File: ttnn/cpp/ttnn/operations/eltwise/unary/unary_composite.hpp [hunk 1]
```diff
@@ -1,120 +1,45 @@
...
-        int32_t param1,
-        const std::optional<MemoryConfig> &memory_config = std::nullopt) {
-        auto output_memory_config = memory_config.value_or(input_tensor.memory_config());
-        return OpHandler<unary_comp_op_type>::handle(input_tensor, param1, output_memory_config);
-        }
+        std::optional<Tensor> optional_output_tensor = std::nullopt);
 };
 
-struct ExecuteRdiv {
+struct RdivOperation {
```
Comment (dmakoviichuk-tt) [inline ttnn/cpp/ttnn/operations/eltwise/unary/unary_composite.hpp:42]: why do you need to have RdivOperation separately?
Is it because of the round_mode?

**Comment by ayerofieiev-tt** [comment]
Will redo to functions registration later

**Pull request closed without merging**